syntax = "proto3";

service Birthright {
   rpc createUpdateBirthrightRule (birthrightRule) returns (birthrightRuleMessage) {}
   rpc readBirthrightRule (readBirthrightData) returns (birthrightData) {}
   rpc deleteBirthrightRule (deleteBirthrightData) returns (deleteBirthrightMessage) {}
   rpc getBirthrightAccess (getAccess) returns (birthrightItemsArray) {}
}

message birthrightRule {
    string status = 1;
    string entitlement = 2;
    map<string, string> rule = 3;
}

message birthrightRuleMessage {
    string birthright_rule_id = 1;
    string status_message = 2;
}

message readBirthrightData {
    string birthright_rule_id = 1;
}

message birthrightData {
    string birthright_rule_id = 1;
    string status = 2;
    string access_catalog_id = 3;
    map<string, string> rule = 4;
}

message deleteBirthrightData {
    string birthright_rule_id = 1;
}

message deleteBirthrightMessage {
    string status_message = 1;
}

message getAccess {
    string identity_id = 1;
}

message birthrightItems {
    string access_data = 1;
}

message birthrightItemsArray {
    repeated birthrightItems items = 1;
}

// python -m grpc_tools.protoc --proto_path=. ./birthright.proto --python_out=. --grpc_python_out=.